pkg/virt-controller/watch/vmi.go

```diff
@@ -634,6 +607,65 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 	return nil
 }
 
+func (c *VMIController) syncReadyConditionFromPod(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) {
+	conditionManager := controller.NewVirtualMachineInstanceConditionManager()
+
+	vmiReadyCond := conditionManager.GetCondition(vmi, virtv1.VirtualMachineInstanceReady)
+	if vmiReadyCond == nil {
+		newCond := virtv1.VirtualMachineInstanceCondition{Type: virtv1.VirtualMachineInstanceReady}
+		vmi.Status.Conditions = append(vmi.Status.Conditions, newCond)
+		vmiReadyCond = &vmi.Status.Conditions[len(vmi.Status.Conditions)-1]
+	}
+
+	setVMICondition := func(status k8sv1.ConditionStatus, reason, message string, lastProbeTime, lastTransitionTime v1.Time) {
+		if vmiReadyCond.Status == status && vmiReadyCond.Reason == reason {
+			return
+		}
+
+		vmiReadyCond.Status = status
+		vmiReadyCond.Reason = reason
+		vmiReadyCond.Message = message
+		vmiReadyCond.LastProbeTime = lastProbeTime
+		vmiReadyCond.LastTransitionTime = lastTransitionTime
+
+		logMessage := "marking VMI as not ready"
+		if status == k8sv1.ConditionTrue {
+			logMessage = "Marking VMI as ready"
+		}
+		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, reason, logMessage)
+		log.Log.Object(vmi).Infof(logMessage + ":" + message)
+	}
+
+	// Keep PodReady condition in sync with the VMI
+	if pod == nil || isTempPod(pod) {
+		setVMICondition(k8sv1.ConditionFalse,
+			virtv1.PodConditionMissingReason,
+			"virt-launcher pod does not exist",
+			v1.Now(),
+			v1.Now())
+
+	} else if isPodDownOrGoingDown(pod) {
+		setVMICondition(k8sv1.ConditionFalse,
+			virtv1.PodTerminatingReason,
+			"virt-launcher pod is terminating",
+			v1.Now(),
```

Comment:
can `v1.Now()` just get called once in this function, stored in a var, and used everywhere we need to represent `now` as a time stamp on a condition

Reply:
Sure, will change accordingly.